spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigMapPropertySourceLocator.java

```diff
@@ -78,24 +82,25 @@ private MapPropertySource getMapPropertySourceForSingleConfigMap(ConfigurableEnv
 			NormalizedSource normalizedSource) {
 
 		String configurationTarget = this.properties.getConfigurationTarget();
-		return getMapPropertySource(getApplicationName(environment, normalizedSource.getName(), configurationTarget),
-				normalizedSource, configurationTarget, environment);
+		String applicationName = getApplicationName(environment, normalizedSource.getName(), configurationTarget);
+		return getMapPropertySource(applicationName, normalizedSource, configurationTarget, environment);
 	}
 
-	protected abstract MapPropertySource getMapPropertySource(String name, NormalizedSource normalizedSource,
-			String configurationTarget, ConfigurableEnvironment environment);
-
 	private void addPropertySourcesFromPaths(Environment environment, CompositePropertySource composite) {
-		this.properties.getPaths().stream().map(Paths::get).peek(p -> {
-			if (!Files.exists(p)) {
+		properties.getPaths().stream().map(Paths::get).filter(p -> {
+			boolean exists = Files.exists(p);
+			if (!exists) {
 				LOG.warn("Configured input path: " + p
 						+ " will be ignored because it does not exist on the file system");
 			}
-		}).filter(Files::exists).peek(p -> {
-			if (!Files.isRegularFile(p)) {
+			return exists;
+		}).filter(p -> {
+			boolean regular = Files.isRegularFile(p);
+			if (!regular) {
 				LOG.warn("Configured input path: " + p + " will be ignored because it is not a regular file");
 			}
-		}).filter(Files::isRegularFile).forEach(p -> {
```

Comment:
there is a big difference between `Stream::forEach` and `List::forEach`. Again, there are many posts from Stuart Marks, Brian Goetz and the like that discuss this in detail. Overall, this can be refactored easily here